homeassistant/components/sensor/whois.py

```diff
@@ -0,0 +1,126 @@
+"""
+Get WHOIS information for a given host.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.whois/
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.const import STATE_UNKNOWN
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['pythonwhois==2.4.3']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_HOST = 'hosts'
+
+ATTR_NAME_SERVERS = 'name_servers'
+ATTR_REGISTRAR = 'registrar'
+ATTR_UPDATED = 'updated'
+ATTR_EXPIRES = 'expires'
+
+SCAN_INTERVAL = timedelta(hours=24)  # WHOIS info is very slow moving
+# We also want to prevent DOS / TOS breaking; One request per domain
+# every 24 hours shouldn't count as "high volume"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): vol.All(cv.ensure_list, [cv.string]),
+})
+
+
+def setup_platform(hass, config, add_devices, disovery_info=None):
+    """Set up the WHOIS sensor."""
+    hostnames = config.get(CONF_HOST)
+
+    devices = []
+
+    for hostname in hostnames:
+        devices.append(WhoisSensor(hass, hostname))
```

Comment:
You should make sure that the domain exists before setting up the platform otherwise the users will end up if a non-working sensor.